pmda/rmsf.py

```diff
@@ -0,0 +1,158 @@
+        sumsquares += (k / (k+1)) * (agroups.positions - mean) ** 2
+        mean = (k * mean + agroups.positions) / (k + 1)
+        return sumsquares, mean
+
+    def run(self,
+                start=None,
+                stop=None,
+                step=None,
+                n_jobs=-1,
+                n_blocks=1,
+                cutoff=15.0):
+            """Perform the calculation
+            Parameters
+            ----------
+            start : int, optional
+                start frame of analysis
+            stop : int, optional
+                stop frame of analysis
+            step : int, optional
+                number of frames to skip between each analysed frame
+            n_jobs : int, optional
+                number of tasks to start, if `-1` use number of logical cpu cores.
+                This argument will be ignored when the distributed scheduler is
+                used
+            """
+            # are we using a distributed scheduler or should we use
+            # multiprocessing?
+            scheduler = dask.config.get('scheduler', None)
+            if scheduler is None:
+                # maybe we can grab a global worker
+                try:
+                    scheduler = dask.distributed.worker.get_client()
+                except ValueError:
+                    pass
+
+            if n_jobs == -1:
+                n_jobs = cpu_count()
+
+            # we could not find a global scheduler to use and we ask for a single
+            # job. Therefore we run this on the single threaded scheduler for
+            # debugging.
+            if scheduler is None and n_jobs == 1:
+                scheduler = 'single-threaded'
+
+            # fall back to multiprocessing, we tried everything
+            if scheduler is None:
+                scheduler = 'multiprocessing'
+
+            scheduler_kwargs = {'scheduler': scheduler}
+            if scheduler == 'multiprocessing':
+                scheduler_kwargs['num_workers'] = n_jobs
+
+            self._indices = self._atomgroup.indices
+            self._ind_slices = np.array_split(self._indices, n_blocks)
+
+            with timeit() as total:
+                with timeit() as prepare:
+                    self._prepare()
+            start, stop, step = self._trajectory.check_slice_indices(
+                start, stop, step)
+            self._frames = range(start, stop, step)
+            blocks = []
+            for ind_slice in self._ind_slices:
+                task = delayed(
+                    self._dask_helper, pure=False)(
+                             ind_slice,
+                             self._top,
+                             self._traj,
+                             self._frames)
+                blocks.append(task)
+            blocks = delayed(blocks)
+            res = blocks.compute(**scheduler_kwargs)
+            self._results = res
+            self.sumsquares = res[0][0]
```

Comment:
This should be handled `_conclude()`. 

Try hard not having to change `run()`!